CONTRIBUTING.md

```diff
@@ -1,37 +1,82 @@
+* Improve our Documentation, the [Coraza Website Repo](https://github.com/corazawaf/coraza.io) is on Github.
 
-* Open a new GitHub pull request with the patch.
+## Reporting an Issue
 
-* Ensure the PR description clearly describes the problem and solution. Include the relevant issue number if applicable.
+* Security related issues are covered by the [Security Policy](https://github.com/corazawaf/coraza/blob/v2/master/SECURITY.md)
+* Make sure you test against the latest version, it's possible the issue was
+  already fixed. However if you are on an older version of Coraza and feel the
+  issue is critical, please let us know.
+* Check existing [Issues](https://github.com/corazawaf/coraza/issues) (open and closed) to ensure it was not already reported.
+* Provide a detailed description and a reproducible test case in a new [Issue](https://github.com/corazawaf/coraza/issues/new).
+  Be sure to include as much relevant information as possible, a **code sample** or an **test case** demonstrating the fault helps us to reproduce your problem.
 
-* Before submitting, please commit using [pre-commit](https://pre-commit.com/) to validate your PR passes the quality tests
+## Patches
+Did you write a patch that fixes a bug?
 
-#### **Do you intend to add a new feature or change an existing one?**
+* Open a new GitHub pull request which includes your changes.
+* Please include a description which clearly describes the change. Include the relevant issue number if applicable.
+* Before submitting, you can test your code using [pre-commit](https://pre-commit.com/) to validate your pull request will pass our set of tests.
 
-* Suggest your change in the [Discussion](https://github.com/corazawaf/coraza/discussions/categories/ideas) and start writing code.
+## Enhancements
 
+Do you intend to add a new feature or change an existing one?
+* Suggest your change in the [Discussion](https://github.com/corazawaf/coraza/discussions/categories/ideas) and start writing code.
 * Do not open an issue on GitHub until you have collected positive feedback about the change. GitHub issues are primarily intended for bug reports and fixes.
+* There are many TODOs, functionalities, fixes, bug reports, and any help you can provide. Just send your pull request.
+
+Run from the repository root:
+```sh
+egrep -Rin "TODO|FIXME" -R --exclude-dir=vendor *
+```
+
+
+## Questions
+Do you have questions about the source code? Ask any question about how to use Coraza in the [community](https://github.com/corazawaf/coraza/discussions/categories/q-a).
+
+## Testing
+
+Coraza uses Go's built-in test tool. Examples (run from the repository root):
+
+- `go test -v`
+- `go test -v -race ` use to enable the built-in data race detector
+- `go test -run TestDefaultWriters -v ./loggers` run all tests loggers package with name substring `TestDefaultWriters`
 
-#### **Do you have questions about the source code?**
+When a pull request is opened CI will run all tests to verify the change.
```

Comment:
When a pull request is opened,  automation will run performing validations and tests to verify the change.